File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l,r=0,len(nums)-1
        m=min(nums[r],nums[l])
        while(l<=r):
            mid=l+(r-l)//2
            print(mid)
            m=min(nums[mid],m)
            if nums[mid]>=nums[r] and nums[mid]>=nums[l]:
                l=mid+1
            else:
                r=mid-1
        return m

        

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l=0,r=nums.size()-1;
        int mid=-1;
        while(l<=r){
            mid=l+(r-l)/2;
            if(target==nums[mid])
                return mid;
            else if(nums[mid]>nums[r]&&(nums[mid]<target||nums[r]>=target)||nums[mid]<=nums[r]&&target<=nums[r]&&nums[mid]<target)
                l=mid+1;
            else
                r=mid-1;
        }
        return -1;        
    }
};